### PR TITLE
AA/MIN-437 balancing switch

### PR DESCRIPTION
**Pull request by ales-by**
**Description:**

Pool balancing switch

**Changes in storage or API:**

Describe storage structure changes or API changes. Make sure to update the [API docs](https://minterestfinance.atlassian.net/wiki/spaces/MINTEREST/pages/134807557/Minterest+API).

**Checklist**

- [ ] Code is covered with tests
- [ ] Code is covered with benchmarks
- [ ] Fresh benchmark results included

### CODE REVIEW

File: pallets/liquidation-pools/src/lib.rs [hunk 1]
```diff
@@ -204,6 +216,23 @@ pub mod module {
...
 	impl<T: Config> Pallet<T> {
+		/// QA only functional!
+		/// Switch on/off pools balancing
+		/// - `new_state`: true - balancing on, false - off
+		///
+		/// origin should be root.
+		#[pallet::weight(10_000)]
+		#[transactional]
+		pub fn switch_balancing_state(origin: OriginFor<T>, new_state: bool) -> DispatchResultWithPostInfo {
+			let _ = ensure_signed(origin)?;
```
Comment (lut1k) [inline pallets/liquidation-pools/src/lib.rs]:
```suggestion
			ensure_root(origin)?;
```

File: pallets/liquidation-pools/src/lib.rs [hunk 2]
```diff
@@ -427,10 +456,15 @@ pub struct Sales {
...
 	fn _offchain_worker(_now: T::BlockNumber) -> Result<(), OffchainErr> {
-		// Check if we are a potential validator
+		// Check if we are a potential validator and balancing is enabled.
 		if !sp_io::offchain::is_validator() {
 			return Err(OffchainErr::NotValidator);
 		}
+
+		if !Self::pool_balancing_state_storage() {
+			return Err(OffchainErr::PoolsBalancingIsOff);
+		}
```
Comment (lut1k) [inline pallets/liquidation-pools/src/lib.rs]:
```suggestion
		ensure!(Self::pool_balancing_state_storage(), OffchainErr::PoolsBalancingIsOff);
```
this suggestion can be applied for check in lines 460-462.

File: pallets/liquidation-pools/src/tests.rs [hunk 3]
```diff
@@ -53,6 +53,47 @@ fn offchain_worker_balancing_test() {
 	});
 }
 
+#[test]
+fn offchain_worker_balancing_off_test() {
+	// balance ratio = 0.2 for two pools. Price the same.
+	// The offchain worker must send transaction for balancing.
+	// It must change 10_000 ETH to 10_000 DOT
```
Comment (lut1k) [inline pallets/liquidation-pools/src/tests.rs]: In my opinion, the description does not match the body of the unit-test.

File: pallets/liquidation-pools/src/tests.rs [hunk 4]
```diff
@@ -53,6 +53,47 @@ fn offchain_worker_balancing_test() {
...
+	ext.register_extension(OffchainWorkerExt::new(offchain));
+	ext.register_extension(TransactionPoolExt::new(pool));
+
+	ext.execute_with(|| {
+		assert_ok!(TestLiquidationPools::switch_balancing_state(admin(), false));
+		let expected_event = Event::TestLiquidationPools(crate::Event::PoolBalacingStateChanged(false));
+		assert!(System::events().iter().any(|record| record.event == expected_event));
+
+		let result = TestLiquidationPools::_offchain_worker(0);
+		assert_eq!(result, Err(OffchainErr::PoolsBalancingIsOff));
```
Comment (lut1k) [inline pallets/liquidation-pools/src/tests.rs]:
```suggestion
		assert_noop!(
			TestLiquidationPools::_offchain_worker(0),
			OffchainErr::PoolsBalancingIsOff
		);
```

File: pallets/liquidation-pools/src/lib.rs [hunk 5]
```diff
@@ -146,13 +148,23 @@ pub mod module {
...
 	pub type LiquidationPoolDataStorage<T: Config> =
 		StorageMap<_, Twox64Concat, CurrencyId, LiquidationPoolData, ValueQuery>;
 
+	#[pallet::type_value]
+	pub fn BalancingStateDefault<T: Config>() -> bool {
+		true
+	}
+	#[pallet::storage]
+	#[pallet::getter(fn pool_balancing_state_storage)]
+	pub type PoolBalancingStateStorage<T: Config> = StorageValue<_, bool, ValueQuery, BalancingStateDefault<T>>;
```
Comment (lut1k) [inline pallets/liquidation-pools/src/lib.rs]: In my opinion, a more appropriate name should be chosen for this variable. From the name "State" it is not entirely obvious that it as a boolean variable. I suggest something similar: `PoolsBalancingEnabled`.

**Review by lut1k — DISMISSED** [review]

**Review by lut1k — APPROVED** [review]

**Review by msifd — APPROVED** [review]

**Pull request merged**